src/interpreter.rs

```diff
@@ -290,9 +290,18 @@ pub fn evaluate(program: &Program) -> Result<(String, String), Error> {
                     &mut err_output,
                 )?,
                 *line_number,
+                statement_source,
             ),
         };
 
+        let get_statement_source = || -> String {
+            input[statement_source.offset..]
```

Comment:
Hm, sometimes `statement_source` is `&str` and sometimes it seems to be more than just a string. What is it in this case? I would have expected that `statement_source` is the current source line.

Reply:
`statement_source` is a Span. So it is a position and remaining text (if something was not parsed). I think is totally fine to reuse this name, since it totally depends on the context. If it is in the interpreter, then it is just a pointer into source code, since you don't have it anymore. In Error type it is just a String.